packages/gatsby/src/datastore/lmdb/lmdb-datastore.ts

```diff
@@ -61,8 +61,31 @@ function getRootDb(): RootDatabase {
   return rootDb
 }
 
+/* eslint-disable @typescript-eslint/no-namespace */
+declare global {
+  namespace NodeJS {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    interface Global {
+      __GATSBY_OPEN_LMDBS?: Map<string, ILmdbDatabases>
+    }
+  }
+}
+
 function getDatabases(): ILmdbDatabases {
   if (!databases) {
+    // __GATSBY_OPEN_LMDBS tracks if we already opened given db in this process
+    // In `gatsby serve` case we might try to open it twice - once for engines
+    // and second to get access to `SitePage` nodes (to power trailing slashes
+    // redirect middleware). This ensure there is single instance within a process.
+    // Using more instances seems to cause weird random errors.
```

Comment:
this seems new with `lmdb@2` - we have trailing slashes in master and released generally and not seeing it there (with `lmdb-store@1`)

Reply:
Oh, and also to clarify - this module does already prevent opening multiple DBs. The problem with `gatsby serve` that this module is used as-is and then it can be used in bundled graphql-engine, so just there are 2 separate instances of this module alive

Reply:
Oooo good catch!